src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -148,6 +167,14 @@ public function loadRoutes(LoaderInterface $loader)
         try {
             $this->configureRoutes(new RoutingConfigurator($collection, $kernelLoader, $file, $file));
 
+            foreach ($collection as $route) {
+                $controller = $route->getDefault('_controller');
+
+                if (($controller[0] ?? null) === $this) {
```

Comment:
What if `$controller` is not an array but a different callable ?

Reply:
check added